src/raft.c

```diff
@@ -639,35 +640,57 @@ static void clearClientSessions(RedisRaftCtx *rr)
 
 static void timeoutBlockedCommand(RedisRaftCtx *rr, raft_entry_t *entry)
 {
+    RaftReq *req = entryDetachRaftReq(rr, entry);
+
     raft_index_t idx;
-    bool error; /* not used yet */
+    bool error;
 
     int ret = RaftRedisDeserializeTimeout(entry->data, entry->data_len, &idx, &error);
     RedisModule_Assert(ret == RR_OK);
 
     BlockedCommand *bc = getBlockedCommand(idx);
     if (!bc) {
         /* unblock handler called before timeout was applied */
+        RedisModule_ReplyWithLongLong(req->ctx, 0);
+        RaftReqFree(req);
         return;
     }
 
     /* In future might have to handle abort failing, but for plain redis commands this is a panic condition */
     ret = RedisModule_CallReplyPromiseAbort(bc->reply, NULL);
-    RedisModule_Assert(ret == REDISMODULE_OK);
 
-    if (bc->req) {
-        if ((strncasecmp(bc->command, "bzpopmin", 8) == 0) ||
-            (strncasecmp(bc->command, "bzpopmax", 8) == 0) ||
-            (strncasecmp(bc->command, "bzmpop", 6) == 0)) {
-            RedisModule_ReplyWithNullArray(bc->req->ctx);
-        } else {
-            RedisModule_ReplyWithNull(bc->req->ctx);
+    if (ret == REDISMODULE_OK) {
```

Comment:
So, can it return REDISMODULE_ERR now?

Reply:
no real difference from before, just trying to be accurate to API.